packages/smooth_app/lib/pages/product/common/product_list_popup_items.dart

```diff
@@ -216,6 +225,135 @@ class ProductListPopupOpenInWeb extends ProductListPopupItem {
   }
 }
 
+/// Popup menu item for the product list page: export list.
+class ProductListPopupExport extends ProductListPopupItem {
+  @override
+  String getTitle(final AppLocalizations appLocalizations) =>
+      appLocalizations.product_list_export;
+
+  @override
+  IconData getIconData() => Icons.download;
+
+  @override
+  ProductListPopupMenuEntry getEntry() => ProductListPopupMenuEntry.export;
+
+  @override
+  bool isDestructive() => false;
+
+  @override
+  Future<ProductList?> doSomething({
+    required final ProductList productList,
+    required final LocalDatabase localDatabase,
+    required final BuildContext context,
+  }) async {
+    final String listName = ProductQueryPageHelper.getProductListLabel(
+      productList,
+      AppLocalizations.of(context),
+    );
+    final String fileName = _buildFileName(listName);
+    final StringBuffer csv = StringBuffer(
+        'Barcode${ProductListPopupItem.EXPORT_SEPARATOR}Name${ProductListPopupItem.EXPORT_SEPARATOR}Brand\n');
```

Comment:
`StringBuffer` allows concatenating strings.
Please split this one, it's hard to read